src/main/java/framework/DriverManager.java

```diff
@@ -34,28 +47,42 @@ public static DriverManager getInstance() {
     private void initializeDriver(){
         String baseUrl = "https://login.salesforce.com";
 
-        if(browser.equalsIgnoreCase("Firefox")){
-            driver = new FirefoxDriver();
-        }
-        else if(browser.equalsIgnoreCase("Chrome")){
-            System.setProperty("webdriver.chrome.driver", "drivers/chromedriver.exe");
-            driver = new ChromeDriver();
-        }
-        else if(browser.equalsIgnoreCase("IE")){
-            System.setProperty("webdriver.ie.driver", "drivers/IEDriverServer.exe");
-            driver = new InternetExplorerDriver();
-        }
-        else if(browser.equalsIgnoreCase("Safari")){
-            driver = new SafariDriver();
+        if (mode.equalsIgnoreCase("Local")) {
+            if (browser.equalsIgnoreCase("Firefox")) {
+                driver = new FirefoxDriver();
+            } else if (browser.equalsIgnoreCase("Chrome")) {
+                System.setProperty("webdriver.chrome.driver", "drivers/chromedriver.exe");
+                driver = new ChromeDriver();
+            } else if (browser.equalsIgnoreCase("IE")) {
+                System.setProperty("webdriver.ie.driver", "drivers/IEDriverServer.exe");
+                driver = new InternetExplorerDriver();
+            } else if (browser.equalsIgnoreCase("Safari")) {
+                driver = new SafariDriver();
+            }
+        }else if (mode.equalsIgnoreCase("Remote")) {
+            DesiredCapabilities caps = null;
+            if(remoteBrowser.equalsIgnoreCase("Chrome")) {
+                caps = DesiredCapabilities.chrome();
+            }else if (remoteBrowser.equalsIgnoreCase("Firefox")) {
+                caps = DesiredCapabilities.firefox();
+            }
+            caps.setCapability("platform", remotePlatform);
+            caps.setCapability("version", remoteVersion);
+            try {
+                this.driver = new RemoteWebDriver(
+                        new URL("http://" + userName + ":" + key + "@ondemand.saucelabs.com:80/wd/hub"),
+                        caps);
+            }catch(MalformedURLException e){
+                LogManager.getInstance().addErrorLog(this.getClass().getName(), "Error on Initializer on remote environment :", e);
+            }
         }
 
-        driver.manage().timeouts().implicitlyWait(15, TimeUnit.SECONDS);
-        wait = new WebDriverWait(driver,15);
+        driver.manage().timeouts().implicitlyWait(90, TimeUnit.SECONDS);
+        wait = new WebDriverWait(driver, 90);
```

Comment:
It is not a good practice to have high timeouts. Is it necessary?

Consider using a variable for that.
Review your constants in the project.